src/Symfony/Component/Security/Http/AccessToken/OidcAccessTokenHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\Exception\InvalidOidcUserException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * The token handler validates and retrieves the user identifier from an OIDC server using the token.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class OidcAccessTokenHandler implements AccessTokenHandlerInterface
```

Comment:
The spec as well as most providers ([including Google](https://developers.google.com/identity/openid-connect/openid-connect#validatinganidtoken)) recommend using this trick only in development or for debugging because an extra HTTP request is required.

Maybe could name it `OidcUserInfoTokenHandler` or something like that, to reflect that it's not the recommended way to do it, and give us the opportunity to add a JWT-based handler if we integrate or recommend a JWT library at some point?

Reply:
And since I don't seen any User object being constructed, will an additional call to the database (or somewhere else) be made by the UserProvider to get the User? Because with JWT, that shouldn't be needed either and the User object can be constructed from the parsed token (which would require a library). I'd also agree to hint this towards `OidcUserInfoTokenHandler`

Reply:
👍 for renaming to `OidcUserInfoTokenHandler` for future-proofness